frontend/app/views/spree/products/show.html.erb

```diff
@@ -96,14 +96,14 @@
       <%= render partial: 'spree/shared/no_product_available' %>
     </div>
   </div>
-
-  <% if related_products.any? %>
-    <div class="product-details-related">
-      <p class="font-weight-bold text-center text-uppercase product-details-carousel-text"><%= Spree.t('pdp.you_may_also_like') %></p>
-
-      <%= render 'spree/shared/carousel_4_products', id: 'related-products-carousel', products: related_products %>
-    </div>
-  <% end %>
 <% end %>
 
 <%= products_structured_data([@product]) %>
+
+<% if @product.respond_to?(:has_related_products?) %>
+  <script>
+    window.onload = function() {
+      Spree.fetchRelatedProductcs('<%= @product.slug %>', '<%= @product.id %>')
```

Comment:
```suggestion
      Spree.fetchRelatedProductcs('<%= @product.slug %>')
```

Spree on the backend will recognize if this is id or slug

https://github.com/spree/spree/blob/7a0450255c594ba985a1b697394b830935a5bac2/frontend/app/controllers/spree/products_controller.rb#L44